### PR TITLE
PIM-5619: Change import system for locales

### PR DESCRIPTION
**Pull request by ClementGautier**
# **/!\ Note for reviewers**
This PR is based on https://github.com/akeneo/pim-community-dev/pull/4093. During your review plz only review the last commit (mine) and not the others.

# Goal
Replace old locales import by 1.4 imports.

Definition Of Done (for Core Developer only)

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Y
| Added Behats                      | N/A
| Changelog updated                 | Y
| Review and 2 GTM                  | todo
| Micro Demo to the PO (Story only) | N/A
| Migration script                  | N/A
| Tech Doc                          | N/A

### CODE REVIEW

File: src/Pim/Component/Catalog/Factory/LocaleFactory.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace Pim\Component\Catalog\Factory;
+
+use Pim\Component\Catalog\Model\LocaleInterface;
+
+/**
+ * Class LocaleFactory
+ *
+ * @author    Clement Gautier <clement.gautier@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
```
Comment (pierallard) [inline src/Pim/Component/Catalog/Factory/LocaleFactory.php]: 2015

File: src/Pim/Component/Catalog/Factory/LocaleFactory.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class LocaleFactory
```
Comment (juliensnz) [inline src/Pim/Component/Catalog/Factory/LocaleFactory.php]: should be dropped

File: src/Pim/Component/Connector/ArrayConverter/Flat/LocaleStandardConverter.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+     * After:
+     * [
+     *     'code'   => 'en_US',
+     * ]
+     */
+    public function convert(array $item, array $options = [])
+    {
+        $this->validator->validateFields($item, ['code']);
+
+        $convertedItem = [];
```
Comment (juliensnz) [inline src/Pim/Component/Connector/ArrayConverter/Flat/LocaleStandardConverter.php]: you can drop that

File: src/Pim/Component/Connector/ArrayConverter/Flat/LocaleStandardConverter.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    /**
+     * @param array  $convertedItem
+     * @param string $field
+     * @param mixed  $data
+     *
+     * @return array
+     */
+    protected function convertField(array $convertedItem, $field, $data)
```
Comment (juliensnz) [inline src/Pim/Component/Connector/ArrayConverter/Flat/LocaleStandardConverter.php]: and that

File: src/Pim/Component/Connector/Processor/Denormalization/LocaleProcessor.php [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        $this->localeConverter = $localeConverter;
+        $this->localeFactory   = $localeFactory;
+        $this->updater         = $updater;
+        $this->validator       = $validator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process($item)
```
Comment (juliensnz) [inline src/Pim/Component/Connector/Processor/Denormalization/LocaleProcessor.php]: should be dropped after my merge

**Comment by jjanvier** [comment]
Last commit is GTM

**Pull request closed without merging**